numpy/ma/tests/test_core.py

```diff
@@ -297,17 +297,28 @@ def test_copy(self):
                         y1._data.__array_interface__)
         self.assertTrue(y1a.mask is y1.mask)
 
-        y2 = array(x1, mask=m)
+        y2 = array(x1, mask=m3)
```

Comment:
Also, what kind of explanation would you like here? Are we looking for just a comment explaining the two use cases? Should I break them into two separate tests?